android-bindings/src/bindings.rs

```diff
@@ -321,7 +321,9 @@ pub unsafe extern "C" fn Java_com_mobilecoin_lib_Amount_init_1jni(
 
         // FIXME #1595: We should get a masked token id also, here we default to
         // 0 bytes, which is backwards compatible
-        let masked_amount = MaskedAmount {
+        // FIXME: This should probably be VersionedMaskedAmount instead, and we
+        // should pass enough data to decide whether to use V1 or V2
```

Comment:
What data would be passed to provide context? 

Potentially some amount of key material like Txout pubkey into a ::new() method for masked amount which can then do a reconstruction of the shared secret to see if it was derived from the original TxOut shared secret or the v2 shared secret? That would be a heavier, but unambigous way to do it?

Perhaps there's a cheaper way of doing that can do an automatic tell without having to pass in something explicitly telling which version to use (which would be operator-error prone)?

Reply:
i think there's no way that can work

they are going to get from the proto they receive data about whether this is masked amount v1 or masked amount v2.

a guess and check operation will require at least one extra elliptic curve point operation, so that's a pretty massive computational overhead, when it's easy enough for them to just pass the data they got from the proto (or, make a proto-oriented API)